### PR TITLE
Fix BTS-1435

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix https://arangodb.atlassian.net/browse/BTS-1435

- [x] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [x] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.11: https://github.com/arangodb/arangodb/pull/19166
  - [x] Backport for 3.10: https://github.com/arangodb/arangodb/pull/19167
  - [ ] Backport for 3.9: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [x] GitHub issue / Jira ticket: https://arangodb.atlassian.net/browse/BTS-1435
- [ ] Design document:

### CODE REVIEW

File: arangod/Aql/Expression.cpp [hunk 1]
```diff
@@ -353,12 +353,19 @@ void Expression::initAccessor() {
...
 
-  TRI_ASSERT(member->type == NODE_TYPE_REFERENCE);
-  auto v = static_cast<Variable const*>(member->getData());
+  if (member->type != NODE_TYPE_REFERENCE) {
+    // the accessor accesses something else than a variable/reference.
+    // this is something we are not prepared for. so fall back to a
+    // simple expression instead
+    _type = SIMPLE;
+  } else {
+    TRI_ASSERT(member->type == NODE_TYPE_REFERENCE);
```
Comment (Dronplane) [inline arangod/Aql/Expression.cpp:362]: Does this assert makes sense?

File: tests/js/server/aql/aql-remove-regression.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                                         "name": u.name,
+                                         "isActive": u.status == "active"
+                                       }
+                                     }
+                       RETURN removed.t.UnitTestCollection`;
+      let actual = db._query(query);
+      assertEqual(actual.toArray().length, 0);
+    },
+    testRegressionCrashEmptyRemoveSmaller : function () {
+      const query = `FOR u IN ${cname}
```
Comment (markuspf) [inline tests/js/server/aql/aql-remove-regression.js:66]: If we really wanted to we could simplify this test even further; not really necessary though

```suggestion
      const query = `FOR u IN 1..2
```

**Review by markuspf — APPROVED** [review]
LGTM

**Review by MBkkt — APPROVED** [review]

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**